### PR TITLE
Add e2e webhook tests

### LINKED ISSUE #610: Add e2e to test webhooks
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
We should test the webhooks are hooked up correctly. Create a few invalid ScyllaClusters and make sure the apiserver denies their creation and updates. The full coverage is enough as a unit tests, but we need to be sure the admission is on and working.

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:**
- Add e2e webhook tests

**Which issue is resolved by this Pull Request:**
Resolves #610

### CODE REVIEW

**Review by tnozicka — CHANGES_REQUESTED** [review]
each entry in `DescribeTable` will result in a `It` block which means a dedicated namesapce is created. I din't think that's necessary when the objects aren't actually created. I'd think you could just do:
- 1 `It` block
- try creation
- expect failure with error code == N
- try another created
- expect failure with error code == N
- ...
- create a basic scyllacluster
- try update
- expect failure with error code == N
- try another update
- expect failure with error code == N
- ...

**Comment by rzetelskik** [comment]
> * expect failure with error code == N

@tnozicka should we actually compare the errors instead of just checking if they occurred? We are already comparing them in the unit tests

**Comment by tnozicka** [comment]
I initially thought about just the HTTP status code, but now I think we should also check the whole message/object. Not as extensively as in unit test where we should have the full matrix coverage, but just one or 2 to make sure the error propagates into the API response

**Comment by rzetelskik** [comment]
> I initially thought about just the HTTP status code, but now I think we should also check the whole message/object. Not as extensively as in unit test where we should have the full matrix coverage, but just one or 2 to make sure the error propagates into the API response

I'll wait for #695 to merge before continuing with this one then

File: test/e2e/set/scyllacluster/scyllacluster_webhooks.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+		)
+		o.Expect(err).To(o.Equal(&errors.StatusError{ErrStatus: metav1.Status{
+			Status:  "Failure",
+			Message: `admission webhook "webhook.scylla.scylladb.com" denied the request: ScyllaCluster.scylla.scylladb.com "basic" is invalid: spec.repository: Forbidden: repository change is currently not supported, old=docker.io/scylladb/scylla, new=new-repo`,
+			Reason:  "Invalid",
+			Details: &metav1.StatusDetails{
+				Name:  "basic",
+				Group: "scylla.scylladb.com",
+				Kind:  "ScyllaCluster",
+				UID:   "",
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply test/e2e/set/scyllacluster/scyllacluster_webhooks.go:109]: We're using `apierrors.NewInvalid` and it's not setting that field https://github.com/kubernetes/kubernetes/blob/9ad065d8040bd19af97ec49a248e4428e0ef2152/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go#L255-L260

[the remark this answers is not in the document]
Comment (tnozicka) [reply test/e2e/set/scyllacluster/scyllacluster_webhooks.go:110]: I guess we could do it in https://github.com/scylladb/scylla-operator/blob/173064eb8d5450913b8b5f102b17ba2e3ce885bc/pkg/admissionreview/handler.go#L96 but lets leave that out for now

File: test/e2e/set/scyllacluster/scyllacluster_webhooks.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = g.Describe("ScyllaCluster webhooks", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewFramework("scyllacluster")
+
+	g.It("should accept/deny requests", func() {
```
Comment (tnozicka) [inline test/e2e/set/scyllacluster/scyllacluster_webhooks.go:24]: Please clean up the overall sentences. It should read like `ScyllaCluster webhook [It] should forbid invalid requests [By] not allowing a ScyllaCluster with duplicated racks`

Comment (rzetelskik) [reply test/e2e/set/scyllacluster/scyllacluster_webhooks.go]: thanks, fixed

**Review by tnozicka — DISMISSED** [review]
/lgtm
thanks

**Review by zimnx — APPROVED** [review]
lgtm, thanks

**Pull request merged**